app/helpers/cubism_helper.rb

```diff
@@ -2,17 +2,34 @@ module CubismHelper
   include CableReady::StreamIdentifier
 
   def cubicle_for(resource, user, html_options: {}, appear_trigger: :connect, disappear_trigger: nil, trigger_root: nil, exclude_current_user: true, &block)
-    key = "#{block.source_location.join(":")}:#{resource.to_gid}:#{user.to_gid}"
-    digested_id = ActiveSupport::Digest.hexdigest(key)
+    filename, lineno = block.source_location
+    block_location = block.source_location.join(":")
+    resource_user_key = "#{resource.to_gid}:#{user.to_gid}"
+    digested_block_key = ActiveSupport::Digest.hexdigest("#{block_location}:#{resource_user_key}")
+
+    # the store item (identified by block location, resource, and user) might already be present
+    store_item = Cubism.store[digested_block_key] || Cubism::BlockStoreItem.new(
+      block_location: block_location,
+      resource_gid: resource.to_gid.to_s,
+      user_gid: user.to_gid.to_s
+    )
```

Comment:
Perhaps an almost meaningless "optimization", but we don't need to get the `gid` twice for `resource` and `user`:

```suggestion
    resource_gid = resource.to_gid.to_s
    user_gid = user.to_gid.to_s
    resource_user_key = [resource_gid, user_gid].join(":")
    digested_block_key = ActiveSupport::Digest.hexdigest("#{block_location}:#{resource_user_key}")

    # the store item (identified by block location, resource, and user) might already be present
    store_item = Cubism.store[digested_block_key] || Cubism::BlockStoreItem.new(
      block_location: block_location,
      resource_gid: resource_gid,
      user_gid: user_gid
    )
```

Reply:
Thanks! Seemingly I can't commit to an already merged PR though 🙈